WalletWasabi/Hwi/ProcessBridge/HwiProcessBridge.cs

```diff
@@ -1,42 +1,43 @@
 using System;
-using System.Collections.Generic;
 using System.Diagnostics;
 using System.IO;
-using System.Runtime.InteropServices;
-using System.Text;
 using System.Threading;
 using System.Threading.Tasks;
-using WalletWasabi.Helpers;
-using WalletWasabi.Interfaces;
-using WalletWasabi.Logging;
 using WalletWasabi.Microservices;
 
 namespace WalletWasabi.Hwi.ProcessBridge
 {
-	public class HwiProcessBridge : Microservices.ProcessBridge
+	public class HwiProcessBridge : IHwiProcessInvoker
 	{
-		public HwiProcessBridge() : base(MicroserviceHelpers.GetBinaryPath("hwi"))
+		public HwiProcessBridge(ProcessInvoker processInvoker)
 		{
+			ProcessPath = MicroserviceHelpers.GetBinaryPath("hwi");
+			ProcessInvoker = processInvoker;
```

Comment:
Every time you create `HwiProcessBridge` it is called like `new HwiProcessBridge(new ProcessInvoker());`
Is there a reason to not create ProcessInvoker inside?

Reply:
At this point, it can be inside. But generally, it's better to pass dependencies via constructor. 

(We decided to go with pure DI. So this way the construction of `HwiProcessBridge` looks strange, but when you use a DI container that creates those things for you then those instantiations are gone as you attempt to use objects in singleton mode often (i.e. you want just once instance of `HwiProcessBridge`)).